src/components/Convert/routes/paraswap.ts

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import { config, NETWORK_NAME_TO_CHAIN_ID, TokenInfo } from "../../..";
+import { BigNumber, ethers } from "ethers";
+import { getApiFeeAsPercentage } from "../getApiFeeAsPercentage";
+import { BASIS_POINTS_DIVISOR } from "../../../config/hlp";
+
+const PARASWAP_FEE_BPS = 100;
+const percentageToParaswapBasisPoints = (percentage: number) =>
+  Math.floor(percentage * PARASWAP_FEE_BPS);
+
+const getParaswapOptimalRate = async (from: TokenInfo, to: TokenInfo, sellAmount: BigNumber) => {
+  const paraswapInstance = Paraswap.constructSimpleSDK({
+    axios,
+    network: from.chainId
+  });
+
+  const optimalRate = await paraswapInstance.getRate({
+    amount: sellAmount.toString(),
+    srcToken: from.address,
+    srcDecimals: from.decimals,
+    destToken: to.address,
+    destDecimals: to.decimals,
+    options: {
+      maxImpact: 100 // 100% as price impact not handled in sdk
+    }
+  });
+  return optimalRate;
+};
+
+// paraswap can be used for any network
+const paraswapWeight = async () => PARASWAP_WEIGHT;
+
+const paraswapQuote = async (input: ConvertQuoteRouteArgs): Promise<RawQuote> => {
+  // paraswap also uses 0xeee...eee as ETH address, so no conversion required
+  const [priceQuote, paraswapApiFeePercentage] = await Promise.all([
+    getParaswapOptimalRate(input.fromToken, input.toToken, input.sellAmount),
+    getApiFeeAsPercentage(input.fromToken.address, input.toToken.address)
+  ]);
+
+  return {
+    allowanceTarget: priceQuote.contractAddress,
+    buyAmount: priceQuote.destAmount,
+    feeBasisPoints: (paraswapApiFeePercentage / 100) * BASIS_POINTS_DIVISOR,
+    feeChargedBeforeConvert: false,
+    gas: +priceQuote.gasCost,
+    sellAmount: input.sellAmount.toString()
+  };
+};
+
+const paraswapTransaction = async (
+  input: ConvertTransactionRouteArgs
+): Promise<ethers.PopulatedTransaction> => {
+  const [paraswapApiFeePercentage, optimalRate, userAddress] = await Promise.all([
+    getApiFeeAsPercentage(input.fromToken.address, input.toToken.address),
+    getParaswapOptimalRate(input.fromToken, input.toToken, input.sellAmount),
+    input.signer.getAddress()
+  ]);
+
+  const paraswapInstance = Paraswap.constructSimpleSDK({
+    axios,
+    network: NETWORK_NAME_TO_CHAIN_ID[input.network]
+  });
+
+  const tx = await paraswapInstance.buildTx(
+    {
+      srcToken: input.fromToken.address,
+      destToken: input.toToken.address,
+      srcAmount: input.sellAmount.toString(),
+      destAmount: input.minOut.toString(),
+      userAddress,
+      receiver: input.receivingAccount,
+      priceRoute: optimalRate,
+      partnerAddress: config.convert.feeAddress,
+      partnerFeeBps: percentageToParaswapBasisPoints(paraswapApiFeePercentage)
+    },
+    {
+      // paraswap is incredibly sensitive to tiny rate changes, and will throw
+      // if the price has slightly changed. However, this is handled in
+      // minOut (field destAmount above), so it can be ignored safely

```

Comment:
I would suggest removing the adjectives
```suggestion
      // paraswap is sensitive to rate changes, and will throw
      // if the price has changed. However, this is handled in
      // minOut (field destAmount above), so it can be ignored safely
```